src/concepts/components/properties.md

```diff
@@ -1,26 +1,551 @@
+#                 <>
+#                     <button onclick=self.link.callback( |_| Msg::Click() )>{"Click here!"} </button>
+#                 </>
+#             }
+#         } else {
+#             html! {
+#                 <>
+#                     <h1>{format!("Hello {}",self.name)}</h1>
+#                 </>
+#             }
+#         }
+#     }
+# }
+
+```
+
+> Here, to simply extend the previous example, we `.clone()` the `props` argument. It may not be needed in your code
+
+## Mandatory property
+
+In this example the property have to be defined using the component. If it's omitted a compilatioin error is raised. The error looks like:
+
+> no method named `build` found for struct `components::comp4::PropsBuilder<...PropsBuilderStep_missing_required_prop_name>` in the current scope
+> method not found in `...::PropsBuilder<...PropsBuilderStep_missing_required_prop_name>`rustc(E0599)
+comp4.rs(14, 10): method `build` not found for this`
+
+## Optional property
+
+Property can be defined optional just adding `#[prop_or_default]` on the property. In that case the property value will be initialized by the default Rust type value.
+
+```rust
+# use yew::prelude::*;
+#
+# pub struct UseOfPropertyComponent {
+#     link: ComponentLink<Self>,
+#     props: Props,
+#     name: String,
+#     show_message: bool,
+# }
+#
+# pub enum Msg {
+#     Click(),
+# }
+
+// ...
+#[derive(Properties, Clone, PartialEq)]
+pub struct Props{
+    #[prop_or_default]
+    pub name: String,
+}
+// ...
+
+# impl Component for UseOfPropertyComponent {
+#     type Message = Msg;
+#     type Properties = Props;
+#
+#     fn create(props: Self::Properties, link: ComponentLink<Self>) -> Self {
+#         Self {
+#             link,
+#             props: props.clone(),
+#             name: props.name.into(),
+#             show_message: false,
+#         }
+#     }
+#
+#     fn update(&mut self, msg: Self::Message) -> ShouldRender {
+#         match msg {
+#             Msg::Click() => self.show_message = true,
+#         }
+#         true
+#     }
+#
+#     fn change(&mut self, props: Self::Properties) -> ShouldRender {
+#         if self.props != props {
+#             self.props = props;
+#             true
+#         } else {
+#             false
+#         }
+#     }
+#
+#     fn view(&self) -> Html {
+#         if !self.show_message {
+#             html! {
+#                 <>
+#                     <button onclick=self.link.callback( |_| Msg::Click() )>{"Click here!"} </button>
+#                 </>
+#             }
+#         } else {
+#             html! {
+#                 <>
+#                     <h1>{format!("Hello {}",self.name)}</h1>
+#                 </>
+#             }
+#         }
+#     }
+# }
+
+
+
+```
+
+In that case we will just say "Hello" ;-)
+
+## Optional property with component default value
+
+Property can be defined optional but adding a default component value adding `#[prop_or_(your_default_value)]` on the property.
```

Comment:
"Property" => "a property"
"can be defined optional but adding a default component value..." => "can be defined as an optional property. In this case, it becomes necessary to define a default component value. Yew will automatically use this value if it is not provided when the component is initialized. Such properties should be annotated with the `#[prop_or_(default_value)]` attribute where `default_value` specifies the value which Yew should use."